src/main/java/edu/hm/hafner/analysis/parser/CodeCheckerParser.java

```diff
@@ -0,0 +1,59 @@
+package edu.hm.hafner.analysis.parser;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+
+
+import edu.hm.hafner.analysis.Issue;
+import edu.hm.hafner.analysis.IssueBuilder;
+import edu.hm.hafner.analysis.LookaheadParser;
+import edu.hm.hafner.analysis.Severity;
+import edu.hm.hafner.util.LookaheadStream;
+
+/**
+ * A parser for the clang-tidy static analysis warnings parsed by Codechecker.
+ *
+ * Codechecker parses the *.plist files and converts it into plain text file.
+ *
+ * Better for human readers and for using grep and diff
+ * It also puts the human-readable Severity at the start of a line.
+ *
+ */
+public class CodeCheckerParser extends LookaheadParser {
+    private static final long serialVersionUID = -3015592762345283582L;
+    private static final String CODE_CHECKER_DEFECT_PATTERN =
+            "^\\[(CRITICAL|HIGH|MEDIUM|LOW)\\] (.+):(\\d+):(\\d+): (.*?) \\[([^\\s]*?)\\]$";
```

Comment:
Can you please use named groups? Example: https://github.com/jenkinsci/analysis-model/blob/master/src/main/java/edu/hm/hafner/analysis/parser/PyLintParser.java

Then the code is more readable.